core/trino-main/src/main/java/io/trino/connector/informationschema/InformationSchemaPageSource.java

```diff
@@ -280,11 +282,23 @@ private void addTablesRecords(QualifiedTablePrefix prefix)
     {
         Set<SchemaTableName> tables = listTables(session, metadata, accessControl, prefix);
         Set<SchemaTableName> views = listViews(session, metadata, accessControl, prefix);
+        Set<SchemaTableName> mviews = listMaterializedViews(session, metadata, accessControl, prefix);
+        Set<SchemaTableName> storageTables = listStorageTab(session, metadata, accessControl, prefix);
         // TODO (https://github.com/trinodb/trino/issues/8207) define a type for materialized views
 
         for (SchemaTableName name : union(tables, views)) {
             // if table and view names overlap, the view wins
-            String type = views.contains(name) ? "VIEW" : "BASE TABLE";
+            String type = "BASE TABLE";
+            if (views.contains(name)) {
+                type = "VIEW";
+            }
+            else if (mviews.contains(name)) {
+                type = "MATERIALIZE VIEW";
+            }
+            else if (storageTables.contains(name)) {
+                type = "STORAGE TABLE";
```

Comment:
I don't think we can give a separate type to storage table of MVs unless the ANSI SQL spec for information schema allows it.

Reply:
base table and mv table is different.
show tables command will print storage table, but i don't  think this table should show

Reply:
from tables perspective, mv's storage table is a table just as any other table.

Reply:
you are right.
But from the user's point of view， they didn't create the storage tables.  so these are transparent to users. they not need to find those table when used "show tabales" command;

Reply:
> But from the user's point of view， they didn't create the storage tables.

It doesn't matter who created a table and why.

> they not need to find those table when used "show tabales" command;

i'd suggesting filing up an issue for this to facilitate a discussion.
i will happily explain why i think we should not bar users from accessing the tables storing mv information directly.